pkgs/development/tools/misc/dotultimate/default.nix

```diff
@@ -0,0 +1,241 @@
+            ''
+              mkdir -p $out/{bin,lib/$pname}
+              cp -a . $out/lib/$pname
+
+              ${lib.getExe extract-dotnet-resources} "${iconHolder}" "*/ProductIcon"
+              ${lib.getExe' icoutils "icotool"} -x */ProductIcon
+              for f in ProductIcon_*.png; do
+                res=$(basename "$f" | cut -d "_" -f3 | cut -d "x" -f1-2)
+                install -vD "$f" "$out/share/icons/hicolor/$res/apps/$pname.png"
+              done
+            ''
+        )
+        + ''
+          runHook postInstall
+        '';
+
+      dontWrapGApps = true;
+
+      fixupPhase = lib.optionalString stdenv.hostPlatform.isLinux ''
+        runHook preFixup
+
+        wrapDotnetProgram() {
+            makeWrapper "$1" "$2" \
+                --suffix "LD_LIBRARY_PATH" : "${lib.makeLibraryPath avaloniaDependencies}" \
+                --set DOTNET_ROOT "${dotnet-runtime}" \
+                --prefix PATH : "${dotnet-runtime}/bin" \
+                "''${gappsWrapperArgs[@]}" \
+                "''${makeWrapperArgs[@]}"
+        }
+
+        ${lib.concatMapStringsSep "\n" (executable: ''
+          wrapDotnetProgram "$out/lib/$pname/${systemToJetBrainsRid system}/${executable}" "$out/bin/${executable}"
+        '') executables}
+
+        runHook postFixup
+      '';
+
+      desktopItems = lib.optionals stdenv.hostPlatform.isLinux [
+        (makeDesktopItem {
+          name = finalAttrs.pname;
+          exec = finalAttrs.meta.mainProgram;
+          inherit desktopName;
+          categories = [ "Development" ];
+          icon = finalAttrs.pname;
+          startupWMClass = "JetBrains ${desktopName}";
+        })
+      ];
+
+      meta = {
+        license = lib.licenses.unfree;
+        mainProgram = builtins.head executables;
+        sourceProvenance = with lib.sourceTypes; [
+          binaryNativeCode
+          binaryBytecode
+        ];
+        platforms = builtins.attrNames sources;
+      } // meta;
+
+      passthru.updateScript = writeShellScript "update-jetbrains.${finalAttrs.pname}" ''
+        set -eu -o pipefail
+        PATH="$PATH:${
+          lib.makeBinPath [
+            curl
+            jq
+            common-updater-scripts
+          ]
+        }"
+
+        latest_release=$(curl -s 'https://data.services.jetbrains.com/products/releases?code=${code}&latest=true&type=release' | jq -r '.${code}[0]')
+        version=$(jq -r '.version' <<< "$latest_release")
+
+        function update_system() {
+          local system="$1"
+          local jetbrains_platform="$2"
+
+          local download=$(jq -r ".downloads[\"$jetbrains_platform\"]" <<< "$latest_release")
+          local download_url=$(jq -r '.link' <<< "$download")
+          local download_sha256=$(curl -s "$(jq -r ".checksumLink" <<< "$download")" | cut -d ' ' -f1)
+
+          update-source-version "$UPDATE_NIX_ATTR_PATH" "$version" "$download_sha256" "$download_url" \
+            --system="$system" --ignore-same-version
```

Comment:
`--system="$system"` does not really work – for example, it is not generally possible to use it to update Darwin source on Linux. I really need to remove it.

Instead, I would recommend exposing the sources in `passthru` and then updating it as attributes, for example: https://github.com/NixOS/nixpkgs/blob/ad8d476b53bd39c93b33d6542eade24230114ac8/pkgs/by-name/gk/gk-cli/package.nix#L45-L115

Reply:
> `--system="$system"` does not really work

It's [used quite a lot in nixpkgs](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20%22--system%3D%22&type=code), are all of those wrong? 

> it is not generally possible to use it to update Darwin source on Linux. 

It's working for me, so I'm curious when would it not work? (except IFD I guess)
 

I'm getting both the url and hash from the updater api anyway so the way to go might be just generating a json file? Or is there a way to modify values in a nix file more semantically without relying on sed?

Reply:
Hmm, might be the case that it works when a hash is specified, since that likely only needs eval. When hash is not provided, Nix will need to build the source derivation so that will fail without compatible builders.